webapp/src/components/common/Form/index.tsx

```diff
@@ -255,113 +259,106 @@ function Form<TFieldValues extends FieldValues, TContext>(
   // API
   ////////////////////////////////////////////////////////////////
 
-  const registerWrapper = useCallback<UseFormRegisterPlus<TFieldValues>>(
-    (name, options) => {
-      if (options?.onAutoSubmit) {
-        fieldAutoSubmitListeners.current[name] = options.onAutoSubmit;
+  const formApiShared = useMemo(
+    () => {
+      if (!isAutoSubmitEnabledRef.current) {
+        return formApiOriginal;
       }
 
-      const newOptions: typeof options = {
-        ...options,
-        onChange: (event: unknown) => {
-          options?.onChange?.(event);
-          if (isAutoSubmitEnabledRef.current) {
-            if (
-              isSubmittingRef.current &&
-              !fieldsChangeDuringAutoSubmitting.current.includes(name)
-            ) {
-              fieldsChangeDuringAutoSubmitting.current.push(name);
+      const registerWrapper: UseFormRegisterPlus<TFieldValues> = (
+        name,
+        options
+      ) => {
+        if (options?.onAutoSubmit) {
+          fieldAutoSubmitListeners.current[name] = options.onAutoSubmit;
+        }
+
+        const newOptions: typeof options = {
+          ...options,
+          onChange: (event: unknown) => {
```

Comment:
If possible, try to provide a more specific type for the `event` parameter like `React.ChangeEvent<HTMLInputElement>`

Reply:
No it will break, the implementation in the lib is `onChange?: (event: any) => void;`.
The convention is to used `unknown`  instead of `any` if it's possible.